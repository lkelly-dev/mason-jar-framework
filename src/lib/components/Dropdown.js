import React from 'react';
import { cx } from 'emotion';
import { Button } from './Button';

const Dropdown = props => {
  const { children, className, ...rest } = props;
  const dropdownClass = cx('dropdown', className);
  return (
    <div className={dropdownClass} {...rest}>
      {children}
    </div>
  );
};

const DropdownToggle = props => {
  const { children, className, ...rest } = props;
  const dropdownToggleClass = cx('dropdown-toggle', className);
  return (
    <Button
      className={dropdownToggleClass}
      type="button"
      data-toggle="dropdown"
      aria-haspopup="true"
      aria-expanded="false"
      {...rest}
    >
      {children}
    </Button>
  );
};

const DropdownFeatherToggle = props => {
  const { className, ...rest } = props;
  const dropdownToggleClass = cx('dropdown-ellipses', 'dropdown-toggle', className);
  return (
    <Button
      type="button"
      className={dropdownToggleClass}
      data-toggle="dropdown"
      aria-haspopup="true"
      aria-expanded="false"
      style={{ border: 'none', cursor: 'pointer' }}
      {...rest}
    >
      <i className="fe fe-more-vertical" />
    </Button>
  );
};

const DropdownMenu = props => {
  const { children, className, ...rest } = props;
  const dropdownMenuClass = cx('dropdown-menu', className);
  return (
    <div className={dropdownMenuClass} {...rest}>
      {children}
    </div>
  );
};

const DropdownItem = props => {
  const { children, className, ...rest } = props;
  const dropdownItemClass = cx('dropdown-item', className);
  return (
    <div className={dropdownItemClass} {...rest}>
      {children}
    </div>
  );
};

const DropdownCard = props => {
  const { children, className, ...rest } = props;
  const dropdownItemClass = cx('dropdown-menu', 'dropdown-menu-card', className);
  return (
    <div className={dropdownItemClass} {...rest}>
      {children}
    </div>
  );
};

Dropdown.Toggle = DropdownToggle;
Dropdown.FeatherToggle = DropdownFeatherToggle;
Dropdown.Menu = DropdownMenu;
Dropdown.Item = DropdownItem;
Dropdown.Card = DropdownCard;

export {
  Dropdown,
  DropdownToggle,
  DropdownFeatherToggle,
  DropdownMenu,
  DropdownItem,
  DropdownCard,
};
