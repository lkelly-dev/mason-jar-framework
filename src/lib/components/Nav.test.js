import React from 'react';
import { shallow } from 'enzyme';
import {
  Nav,
  NavTab,
  NavItem,
  NavLink,
  NavDropdown
} from '../index.js';
import { Dropdown } from './Dropdown';

describe('Nav', () => {
  it('Nav matches snapshot', () => {
    const component = shallow(<Nav></Nav>);
    expect(component).toMatchSnapshot();
  })
  it('has expected basic props', () => {
    const children = <div> children </div>;
    const component = shallow(
      <Nav
        what="ever"
        className={ cn }>
        { children }
      </Nav>
    );
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain('nav');
    expect(classNameArray).toContain(cn);
    expect(classNameArray).toHaveLength(2);
    expect(component.prop('what')).toBe('ever');
    expect(component.type()).toBe('nav')
  })

  it('appends data-attributes in debugMode', () => {
    const Context = React.createContext({debugMode: true});
    const child = "text";
    const component = shallow(
      <Nav
        context={ Context }
        className={ cn }
        what="ever">
        { child }
      </Nav>
    );
    expect(component.prop('data-context')).not.toBeDefined();
    expect(component.prop('data-children')).not.toBeDefined();
    expect(component.prop('data-what')).toBe('ever');
    expect(component.prop('data-className')).toBe(cn);
  })
});

describe('NavTab', () => {
  it('NavTab matches snapshot', () => {
    const component = shallow(<NavTab></NavTab>);
    expect(component).toMatchSnapshot();
  })
  it('has expected basic props', () => {
    const overflow = 'overflow',
          children = <div>child</div>;
    const component = shallow(
      <NavTab
        what="ever"
        size={ size }
        overflow={ overflow }
        className={ cn }>
        { children }
      </NavTab>
    );
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain('nav');
    expect(classNameArray).toContain(cn);
    expect(classNameArray).toContain(`nav-tabs-${size}`);
    expect(classNameArray).toContain('nav-overflow');
    expect(classNameArray).toHaveLength(5);
    expect(component.prop('what')).toBe('ever');
    expect(component.type()).toBe('nav')
  })

  it('appends data-attributes in debugMode', () => {
    const Context = React.createContext({debugMode: true});
    const child = 'text',
          overflow = 'overflow';
    const component = shallow(
      <NavTab
        context={ Context }
        size={ size }
        overflow={ overflow }
        className={ cn }
        what="ever">
        { child }
      </NavTab>
    );
    expect(component.prop('data-context')).not.toBeDefined();
    expect(component.prop('data-children')).not.toBeDefined();
    expect(component.prop('data-what')).toBe('ever');
    expect(component.prop('data-size')).toBe(size);
    expect(component.prop('data-overflow')).toBe(overflow);
    expect(component.prop('data-className')).toBe(cn);
  })
});

describe('NavItem', () => {
  it('NavItem matches snapshot', () => {
    const component = shallow(<NavItem></NavItem>);
    expect(component).toMatchSnapshot();
  })
  it('has expected basic props', () => {
    const children = <div> children </div>;
    const component = shallow(
      <NavItem
        what="ever"
        className={ cn }>
        { children }
      </NavItem>
    );
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain('nav-item');
    expect(classNameArray).toContain(cn);
    expect(classNameArray).toHaveLength(2);
    expect(component.prop('what')).toBe('ever');
    expect(component.type()).toBe('div')
  })

  it('appends data-attributes in debugMode', () => {
    const Context = React.createContext({debugMode: true});
    const child = 'text',
          overflow = 'overflow';
    const component = shallow(
      <NavItem
        context={ Context }
        className={ cn }
        what="ever">
        { child }
      </NavItem>
    );
    expect(component.prop('data-context')).not.toBeDefined();
    expect(component.prop('data-children')).not.toBeDefined();
    expect(component.prop('data-what')).toBe('ever');
    expect(component.prop('data-className')).toBe(cn);
  })
});

describe('NavDropdown', () => {
  it('NavDropdown matches snapshot', () => {
    const component = shallow(<NavDropdown></NavDropdown>);
    expect(component).toMatchSnapshot();
  })
  it('has expected basic props', () => {
    const children = <div> children </div>;
    const component = shallow(
      <NavDropdown
        what="ever"
        className={ cn }>
        { children }
      </NavDropdown>
    );
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain('nav-item');
    expect(classNameArray).toContain('dropdown');
    expect(classNameArray).toContain(cn);
    expect(classNameArray).toHaveLength(3);
    expect(component.prop('what')).toBe('ever');
    expect(component.type()).toBe(Dropdown)
  })

  it('appends data-attributes in debugMode', () => {
    const Context = React.createContext({debugMode: true});
    const child = 'text';
    const component = shallow(
      <NavDropdown
        context={ Context }
        className={ cn }
        what="ever">
        { child }
      </NavDropdown>
    );
    expect(component.prop('data-context')).not.toBeDefined();
    expect(component.prop('data-children')).not.toBeDefined();
    expect(component.prop('data-what')).toBe('ever');
    expect(component.prop('data-className')).toBe(cn);
  })
});

describe('NavLink', () => {
  it('NavLink matches snapshot', () => {
    const component = shallow(<NavLink></NavLink>);
    expect(component).toMatchSnapshot();
  })
  it('has expected props', () => {
    const active = 'active',
          children = <div>children</div>;
    const component = shallow(
      <NavLink
        what="ever"
        active={ active }
        className={ cn }>
        { children }
      </NavLink>
    );
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain('nav-link');
    expect(classNameArray).toContain('active');
    expect(classNameArray).toContain(cn);
    expect(classNameArray).toHaveLength(4);
    expect(component.prop('what')).toBe('ever');
    expect(component.type()).toBe('div')
  });
  it('can have custom element', () => {
    const component = shallow(<NavLink as="i"/>);
    expect(component.type()).toBe('i');
  })

  it('appends data-attributes in debugMode', () => {
    const Context = React.createContext({debugMode: true});
    const child = 'text',
          active = 'active';
    const component = shallow(
      <NavDropdown
        context={ Context }
        active={ active }
        className={ cn }
        what="ever">
        { child }
      </NavDropdown>
    );
    expect(component.prop('data-context')).not.toBeDefined();
    expect(component.prop('data-children')).not.toBeDefined();
    expect(component.prop('data-what')).toBe('ever');
    expect(component.prop('data-active')).toBe(active);
    expect(component.prop('data-className')).toBe(cn);
  })
});
