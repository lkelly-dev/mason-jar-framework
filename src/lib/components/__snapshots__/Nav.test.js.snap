// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav matches snapshot 1`] = `
<nav
  className="nav"
/>
`;

exports[`NavDropdown NavDropdown matches snapshot 1`] = `
<Dropdown
  className="nav-item dropdown"
/>
`;

exports[`NavItem NavItem matches snapshot 1`] = `
<div
  className="nav-item"
/>
`;

exports[`NavLink NavLink matches snapshot 1`] = `
<div
  className="nav-link css-1h5x3dy"
/>
`;

exports[`NavTab NavTab matches snapshot 1`] = `
<nav
  className="nav nav-tabs"
/>
`;
